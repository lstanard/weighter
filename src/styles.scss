@import "./scss/globals";

body {
  color: $slate;
  margin: 0;
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen",
    "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue",
    sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

code {
  font-family: source-code-pro, Menlo, Monaco, Consolas, "Courier New",
    monospace;
}

html {
  box-sizing: border-box;
}

*,
*:before,
*:after {
  box-sizing: inherit;
}

// BEGIN TEMP STYLES

.root {
  height: 100%;
}

h1 {
  color: $lime;
  font-family: "Work Sans", sans-serif;
  font-weight: 300;
  margin: 0;
}

h3 {
  margin: 0;
}

h2,
h3 {
  color: $concrete;
}

p {
  color: $slate;
  font-size: 15px;
}

.main {
  display: flex;
  flex-direction: row;
  height: 100%;
  min-height: 100vh;
}

.header {
  margin: 15px 0 30px;
}

.availability-panel {
  width: 70%;
}

.equipment-panel {
  background-color: $jade;
  padding: 15px 30px;
  max-height: 100vh;
  max-width: 30%;
  min-width: 300px;

  p {
    color: $concrete;
  }
}

// END TEMP STYLES
