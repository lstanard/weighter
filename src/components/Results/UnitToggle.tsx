import React, { ReactElement } from "react";
import cn from "classnames";

import { LB, KG } from "../../constants/units";
import {
  useGlobalUnitsContext,
  GlobalUnitsContextValues,
} from "../useGlobalUnitsContext";

import styles from "./UnitToggle.module.scss";

const UnitToggle = (): ReactElement => {
  const { units, setUnits }: GlobalUnitsContextValues = useGlobalUnitsContext();

  return (
    <div className={styles.container}>
      <button
        type="button"
        onClick={(): void => setUnits(LB)}
        className={cn({
          [styles.selected]: units === LB,
        })}
      >
        Pounds
      </button>
      <button
        type="button"
        onClick={(): void => setUnits(KG)}
        className={cn({
          [styles.selected]: units === KG,
        })}
      >
        Kilograms
      </button>
    </div>
  );
};

export default UnitToggle;
