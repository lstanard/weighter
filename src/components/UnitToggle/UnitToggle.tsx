import React, { ReactElement } from "react";
import cn from "classnames";

import { LB, KG } from "../../constants/units";
import {
  useGlobalUnitsContext,
  GlobalUnitsContextValues,
} from "../useGlobalUnitsContext";

import styles from "./UnitToggle.module.scss";

export interface UnitToggleProps {
  className?: string;
}

const UnitToggle = ({ className }: UnitToggleProps): ReactElement => {
  const { units, setUnits }: GlobalUnitsContextValues = useGlobalUnitsContext();

  return (
    <div className={cn(styles.container, className)}>
      <button
        type="button"
        onClick={(): void => setUnits(LB)}
        className={cn({
          [styles.selected]: units === LB,
        })}
      >
        Pounds
      </button>
      <button
        type="button"
        onClick={(): void => setUnits(KG)}
        className={cn({
          [styles.selected]: units === KG,
        })}
      >
        Kilograms
      </button>
    </div>
  );
};

export default UnitToggle;
