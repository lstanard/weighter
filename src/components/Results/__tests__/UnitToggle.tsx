import React from "react";
import { render, fireEvent } from "@testing-library/react";

import UnitToggle from "../UnitToggle";

import styles from "../UnitToggle.module.scss";

describe("UnitToggle", () => {
  it("should render correctly", () => {
    const { getByText, container } = render(<UnitToggle />);
    expect(getByText("Pounds")).not.toBeNull();
    expect(getByText("Kilograms")).not.toBeNull();
    expect(container).toMatchSnapshot();
  });

  it("should select pounds by default", () => {
    const { container } = render(<UnitToggle />);
    const buttons = container.querySelectorAll("button");
    expect(buttons[0].classList).toContain(styles.selected);
    expect(buttons[1].classList).not.toContain(styles.selected);
  });

  // TODO: not sure how to get this test working with useGlobalUnitsContext
  it.skip("should correctly toggle selection", () => {
    const { container, debug } = render(<UnitToggle />);
    const buttons = container.querySelectorAll("button");
    fireEvent.click(buttons[1]);
    debug();
    expect(container).toMatchSnapshot();
  });
});
