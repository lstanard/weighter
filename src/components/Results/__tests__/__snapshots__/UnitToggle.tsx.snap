// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnitToggle should correctly toggle selection 1`] = `
<div>
  <div
    class="container"
  >
    <button
      class="selected"
      type="button"
    >
      Pounds
    </button>
    <button
      class=""
      type="button"
    >
      Kilograms
    </button>
  </div>
</div>
`;

exports[`UnitToggle should render correctly 1`] = `
<div>
  <div
    class="container"
  >
    <button
      class="selected"
      type="button"
    >
      Pounds
    </button>
    <button
      class=""
      type="button"
    >
      Kilograms
    </button>
  </div>
</div>
`;
